sunpy/map/mapbase.py

```diff
@@ -265,7 +265,15 @@ def wcs(self):
         """
         The `~astropy.wcs.WCS` property of the map.
         """
-        w2 = astropy.wcs.WCS(naxis=2)
+        # Construct the WCS based on the FITS header, but don't "do_set" which
+        # analyses the FITS header for correctness.
+        w2 = astropy.wcs.WCS(header=self.fits_header, _do_set=False)
+
+        # If the FITS header is > 2D pick the first 2 and move on.
+        # This will require the FITS header to be valid.
+        if w2.naxis != 2:
```

Comment:
To silently do this seems a bit...questionable. At the very least, shouldn't we tell the user that the higher dimensions are being skipped?

Reply:
I thought we already did that somewhere on map load?